src/repl/scala/tools/nsc/interpreter/ILoop.scala

```diff
@@ -633,28 +634,29 @@ class ILoop(in0: Option[BufferedReader], protected val out: JPrintWriter)
    *  the interpreter and replays all interpreter expressions.
    */
   def require(arg: String): Unit = {
-    class InfoClassLoader extends java.lang.ClassLoader {
-      def classOf(arr: Array[Byte]): Class[_] =
-        super.defineClass(null, arr, 0, arr.length)
-    }
-
     val f = File(arg).normalize
 
-    if (f.isDirectory) {
-      echo("Adding directories to the classpath is not supported. Add a jar instead.")
+    val jarFile = AbstractFile.getDirectory(new java.io.File(arg))
+    if (jarFile == null) {
+      echo(s"Cannot load '$arg'")
       return
     }
 
-    val jarFile = AbstractFile.getDirectory(new java.io.File(arg))
-
     def flatten(f: AbstractFile): Iterator[AbstractFile] =
       if (f.isClassContainer) f.iterator.flatMap(flatten)
       else Iterator(f)
 
     val entries = flatten(jarFile)
-    val cloader = new InfoClassLoader
 
-    def classNameOf(classFile: AbstractFile): String = cloader.classOf(classFile.toByteArray).getName
+    def classNameOf(classFile: AbstractFile): String = {
+      val input = classFile.input
```

Comment:
You guys don't have try with resources? What kind of a language is this, anyway?